### PR TITLE
Support `indices.groups` when compiling named groups in regexps

### PR DESCRIPTION
**Pull request by ptomato**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | None opened
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | Yes (?)
| Tests Added + Pass?      | Yes
| Documentation PR Link    | No
| Any Dependency Changes?  | No
| License                  | MIT

If a RegExp has the `/d` flag (`hasIndices`), match objects should have an extra property `.indices` with an array of the character indices that each subgroup matched. The indices array itself has a property `.groups` giving the matched character indices (or undefined) for each named capturing group.

This doesn't do anything to enable support for the `/d` flag, the `.hasIndices` property, or the returned indices object, if the environment does not support it, but it will populate `.indices.groups` in `_wrapRegExp` if `.indices` is present, so that the transform-named-capturing-groups-regex and proposal-duplicate-named-capturing-groups-regex plugins can take advantage of that support if it exists.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15092"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
I'm marking this as a bugfix because it just adds a missing piece of a proposal we already implemented (duplicate named groups), and it only works in browsers that already support `/d`. It would be a new feature if it implemented full support for `/d`, which we don't have yet.

File: packages/babel-helpers/src/helpers/wrapRegExp.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ export default function _wrapRegExp() {
   BabelRegExp.prototype.exec = function (str) {
     var result = _super.exec.call(this, str);
     if (result) result.groups = buildGroups(result, this);
+    if (this.hasIndices && result) {
```
Comment (nicolo-ribaudo) [inline packages/babel-helpers/src/helpers/wrapRegExp.js:25]: Should we also check if `result.indices` exists, for compat with older engines?

Comment (ptomato) [reply packages/babel-helpers/src/helpers/wrapRegExp.js:25]: I think we only have to check for one of `result.indices` or `this.hasIndices`, not both. I don't think there should be a case where one is truthy and the other isn't (either on engines that don't support `/d`, or on engines that do support it but the regexp doesn't have the `/d` flag.)

`result.indices` is one character shorter than `this.hasIndices`, and it fits well with your suggested code snippet above, which is more elegant than what I wrote. So I'll go with that.

**Comment by nicolo-ribaudo** [comment]
For the failing tests: Node.js 12 and older don't support `/d`, so they throw. You can fix it by:
- Moving the test from `has-indices.js` to `has-indices/exec.js`
- Add an `has-indices/options.json` file that specifies the minimum Node.js version, like https://github.com/babel/babel/blob/672a58660f0b15691c44582f1f3fdcdac0fa0d2f/packages/babel-plugin-transform-named-capturing-groups-regex/test/fixtures/runtime/match-all/options.json

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by JLHwung — APPROVED** [review]

**Pull request merged**